runtime-light/stdlib/string/regex-functions.cpp

```diff
@@ -604,8 +653,17 @@ Optional<string> f$preg_replace(const mixed &pattern, const string &replacement,
   return result;
 }
 
-Optional<string> f$preg_replace(const mixed &pattern, const mixed &replacement, const string &subject, int64_t limit, int64_t &count) noexcept {
-  count = 0;
+Optional<string> f$preg_replace(const mixed &pattern, const mixed &replacement, const string &subject, int64_t limit,
+                                Optional<std::variant<std::monostate, std::reference_wrapper<int64_t>>> opt_count) noexcept {
+  int64_t count{};
+  vk::final_action count_finalizer{[&count, &opt_count]() noexcept {
+    if (opt_count.has_value()) {
+      php_assert(std::holds_alternative<std::reference_wrapper<int64_t>>(opt_count.val()));
+      auto &inner_ref{std::get<std::reference_wrapper<int64_t>>(opt_count.val()).get()};
```

Comment:
NIT: And better syntax for `holds_alternative` and `get`?

Reply:
I didn't understand your comment. Can you explain it a bit?

Reply:
Referring to the previous conversation about type alias or wrapper